src/main/scala/org/allenai/plugins/ScaladocGenPlugin.scala

```diff
@@ -0,0 +1,216 @@
+  /** Extra libraries map, with CoreDependencies java libraries. */
+  lazy val extraJavadocMapSetting = scaladocGenExtraJavadocMap := {
+    Map("slf4j-api" -> url("http://www.slf4j.org/api/"))
+  }
+
+  /** Setting to add API mappings for our Java libraries. This allows the scaladoc generation task
+    * to generate links for these libraries.
+    */
+  lazy val javaApiMappingsSetting = apiMappings ++= {
+    val log = streams.value.log
+    // From the full classpath, find jars with the names given in our doc map.
+    val classpath: Iterable[File] = aggregateFullClasspath.value.flatten map { _.data }
+    val externalJavadocMappings = scaladocGenExtraJavadocMap.value flatMap {
+      case (libraryName, javadocUrl) => {
+        classpath find { file => (file ** s"$libraryName*.jar").get.nonEmpty } map { jar =>
+          jar -> javadocUrl
+        }
+      }
+    }
+    // Add the runtime mapping to the mappings we just generated & return.
+    externalJavadocMappings + (rtJar -> scaladocGenJavadocUrl.value)
+  }
+
+  /** Setting that overrides the doc generation task. This finds all HTML files generated by the doc
+    * task, then subs in the correctly-formatted Javadoc link in place of the badly-formatted
+    * Scaladoc link.
+    */
+  lazy val fixJavadocSetting = doc in ScalaUnidoc := {
+    // Build up a regular expression that matches any of the external javadoc links we have.
+    val externalUrls =
+      scaladocGenJavadocUrl.value +: scaladocGenExtraJavadocMap.value.values.toVector
+    val anyUrl = externalUrls map { url => s"""\\Q$url\\E""" } mkString { "|" }
+    val captureUrlAndFragment = s""""($anyUrl)#([^"]*)"""".r
+
+    // Run the doc task, and get the folder it generated the HTML files into.
+    val docFolder: File = (doc in ScalaUnidoc).value
+    val htmlFiles: Iterable[File] = (docFolder ** "*.html").get
+    // Replace all the Javadoc links in the generated files.
+    htmlFiles foreach { file => 
+      val oldContent: String = IO.read(file)
+      val newContent: String = captureUrlAndFragment.replaceAllIn(oldContent, { regexMatch =>
+        val urlBase = regexMatch.group(1)
+        val scaladocFragment = regexMatch.group(2)
+        val javadocPage = scaladocFragment.replace(".", "/")
+        s"$urlBase?$javadocPage.html"
+      })
+      if (oldContent != newContent) {
+        IO.write(file, newContent)
+      }
+    }
+    // Return the same doc folder from the task.
+    docFolder
+  }
+
+  // The following are helpers to make it easier to add manual mappings for Scala libraries.
+
+  lazy val scaladocGenExtraScaladocMap: SettingKey[Map[String, URL]] = settingKey[Map[String, URL]](
+    "Mappings for any Scala libraries to link in, of library name to Scaladoc URL"
+  )
+  /** The default setting includes libraries from CoreDependencies. */
+  lazy val extraScaladocMapSetting = scaladocGenExtraScaladocMap := {
+    val akkaUrl = url("http://doc.akka.io/api/akka/2.4.0")
```

Comment:
Yes, without overriding this map setting for the appropriate key(s). Note that if we sent akka a patch that added the `info.apiURL` property to their pom.xml, we could remove these. That's (obviously) true of any of these. I might do that real quick . . .